__init__.py

```diff
@@ -171,6 +171,10 @@ def weather_at_location(self, name):
                 "path": "/weather",
                 "query": q
             })
+            # 404 returned as JSON-like string not HTTPError in some instances
+            if type(data) is str and'{"cod":"404"' in data:
+                name = ' '.join(name.split()[:-1])
+                return self.weather_at_location(name)
```

Comment:
Would it be cleaner to override the request method with

  def request(*arg, **kwarg):
    ret = super().request(*arg, **kwarg)
    if isinstance(ret, str) and '{"cod":"404"' in data:
      r = Response()
      r.status_code = 404
      raise HTTPError(data, response=r)

Then the normal 404 handling should catch it and there would be no code duplication...

Reply:
Much nicer!
I integrated it into the existing method, take a look and see what you think.